src/python_testing/TC_DA_1_1.py

```diff
@@ -0,0 +1,225 @@
+                     - Verify that the FabricID for that entry matches the FabricID for TH1"""),
+            TestStep(3, "Factory reset DUT and perform the necessary actions to put the DUT into a commissionable state"),
+            TestStep(4, "Commission DUT to TH2's Fabric"),
+            TestStep(5, "TH2 does a non-fabric-filtered read of Fabrics attribute list from DUT", """
+                     - Verify that there is a single entry in the Fabrics list
+                     - Verify that the FabricID is the same as TH2's FabricID
+                     - Verify that TH2's Fabrics attribute's FabricID is different than from TH1's Fabrics attribute's FabricID"""),
+            TestStep(6, "TH2 does a non-fabric-filtered read of NOCs attribute list from DUT", """
+                     - Verify that there is a single entry in the NOCs list
+                     - Verify that TH2's NOCs entry's public key is different than TH1's NOCs entry's public key"""),
+        ]
+
+        return steps
+
+    def get_new_controller(self) -> ChipDeviceCtrl.ChipDeviceController:
+        new_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        new_fabric_admin = new_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.matter_test_config.fabric_id + 1)
+        new_controller = new_fabric_admin.NewController(paaTrustStorePath=str(self.matter_test_config.paa_trust_store_path))
+        return new_controller
+
+    async def read_nocs(self, dev_ctrl: ChipDeviceCtrl.ChipDeviceController):
+        return await self.read_single_attribute_check_success(
+            dev_ctrl=dev_ctrl,
+            endpoint=0,
+            cluster=Clusters.OperationalCredentials,
+            attribute=Clusters.OperationalCredentials.Attributes.NOCs,
+            fabric_filtered=False)
+
+    async def read_fabrics(self, dev_ctrl: ChipDeviceCtrl.ChipDeviceController):
+        return await self.read_single_attribute_check_success(
+            dev_ctrl=dev_ctrl,
+            endpoint=0,
+            cluster=Clusters.OperationalCredentials,
+            attribute=Clusters.OperationalCredentials.Attributes.Fabrics,
+            fabric_filtered=False)
+
+    async def factory_reset_dut(self, dev_ctrl: ChipDeviceCtrl.ChipDeviceController):
+        # Delete KVS files
+        kvs_files = list(pathlib.Path(".").glob("kvs1*"))
+        for kvs_file in kvs_files:
+            logging.info(f"Deleting KVS file: {kvs_file}")
+            kvs_file.unlink(missing_ok=True)
+
+        # Delete admin storage file
+        admin_storage = pathlib.Path("admin_storage.json")
+        if admin_storage.exists():
+            admin_storage.unlink()
+
+        # Trigger app restart for factory reset
+        restart_flag_file = self.get_restart_flag_file()
+        if restart_flag_file:
+            with open(restart_flag_file, "w") as f:
+                f.write("restart")
```

Comment:
instead of handling the factory reset in this file, can you handle it in the run_python_test file in the monitor_app_restart_requests function? That will let you get the actual KVS file from the --KVS argument and it will let other tests use the same logic. If you check for what's written to the file, you should be able to differentiate between a restart and a reset request ex by writing "reset" instead of "restart" here.